elogia_sale/models/elogia_sale.py

```diff
@@ -0,0 +1,811 @@
+
+    def action_view_moves(self):
+        action = self.env["ir.actions.actions"]._for_xml_id("account.action_move_journal_line")
+        action['domain'] = [('control_id', '=', self.id), ('move_type', '=', 'entry')]
+        action['context'] = {'default_move_type': 'entry', 'view_no_maturity': True,
+                             'default_control_id': self.id,
+                             'default_campaign_elogia_id': self.campaign_id.id
+                             }
+        return action
+
+
+class ObjectiveCampaignMarketing(models.Model):
+    _name = 'objective.campaign.marketing'
+    _inherit = ['portal.mixin', 'mail.thread', 'mail.activity.mixin', 'utm.mixin']
+    _description = "Objective Campaign Marketing"
+    _order = 'id desc'
+
+    campaign_line_id = fields.Many2one('campaign.marketing.line', 'Campaign Line', required=True, ondelete='restrict',
+                                       index=True, copy=False)
+    name = fields.Char(string='Description', required=True, index=True)
+    project_id = fields.Many2one('project.project', 'Project', tracking=1)
+    campaign_id = fields.Many2one('campaign.marketing.elogia', 'Campaign')
+    currency_id = fields.Many2one(related='campaign_id.currency_id', depends=['campaign_id.currency_id'], store=True,
+                                  string='Currency')
+    company_id = fields.Many2one(related='campaign_id.company_id', string='Company', store=True, index=True)
+    client_id = fields.Many2one('res.partner', 'Client', tracking=1)
+    partner_id = fields.Many2one('res.partner', 'Supplier', tracking=1)
+    product_id = fields.Many2one('product.product', string='Product', domain="[('sale_ok', '=', True)]", tracking=1)
+    period = fields.Many2one('period.campaign', 'Period', tracking=1, ondelete='restrict')
+    amount_period = fields.Float('Amount', tracking=1)
+    percentage_fee = fields.Float('% Fee', tracking=1)
+    percentage_margin = fields.Float('% Margin', tracking=1)
+    margin = fields.Float('Margin', tracking=1, compute="calc_amount_margin")
+    type_payment = fields.Selection(selection=[
+        ('spain', 'Spain'),
+        ('client', 'Client'),
+        ('mexico', 'Mexico'),
+    ], string='Payment type', tracking=1)
+    month_actual = fields.Integer('Month')
+    check_deleted = fields.Boolean('To delete')
+
+    def action_read_objective(self):
+        self.ensure_one()
+        return {
+            'name': self.display_name,
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'objective.campaign.marketing',
+            'res_id': self.id,
+        }
+
+    @api.depends('amount_period', 'percentage_margin')
+    def calc_amount_margin(self):
+        for record in self:
+            record.margin = 0
+            if record.amount_period and record.percentage_margin:
+                record.margin = record.amount_period * record.percentage_margin / 100
```

Comment:
la asiganacion debe estar fuera del if, en caso que no entre al if te devolvera un error porque no pudo asiganarle valor al campo computado

Reply:
sorry ya veo que lo asignas arriba, en ese caso estas cambiando el valor dos veces, en mi opinion seria quedaria mejor 
    def calc_amount_margin(self):
        for record in self:
            margin = 0
            if record.amount_period and record.percentage_margin:
               margin = record.amount_period * record.percentage_margin / 100 
            record.margin=margin